lib/view_model/wallet_new_vm.dart

```diff
@@ -15,23 +15,38 @@ import 'package:cw_core/wallet_type.dart';
 import 'package:cake_wallet/view_model/wallet_creation_vm.dart';
 import 'package:cake_wallet/bitcoin/bitcoin.dart';
 import 'package:cake_wallet/haven/haven.dart';
+import 'advanced_privacy_settings_view_model.dart';
 
 part 'wallet_new_vm.g.dart';
 
 class WalletNewVM = WalletNewVMBase with _$WalletNewVM;
 
 abstract class WalletNewVMBase extends WalletCreationVM with Store {
   WalletNewVMBase(AppStore appStore, WalletCreationService walletCreationService,
-      Box<WalletInfo> walletInfoSource,
+      Box<WalletInfo> walletInfoSource, this.advancedPrivacySettingsViewModel,
       {required WalletType type})
       : selectedMnemonicLanguage = '',
         super(appStore, walletInfoSource, walletCreationService, type: type, isRecovery: false);
 
+  final AdvancedPrivacySettingsViewModel advancedPrivacySettingsViewModel;
+
   @observable
   String selectedMnemonicLanguage;
 
   bool get hasLanguageSelector => type == WalletType.monero || type == WalletType.haven;
 
+  int get seedPhraseWordsLength {
+      switch (type) {
+        case WalletType.monero:
+          return 25;
+        case WalletType.ethereum:
+        case WalletType.bitcoinCash:
+          return advancedPrivacySettingsViewModel.seedPhraseLength.value;
+        default:
+          return 24;
+      }
+    }
```

Comment:
I believe this is not the case anymore after introducing polyseed, so let's carefully merge the latest changes and solve the conflicts with respect to the new changes

Reply:
fixed